pkg/installer/installer.go

```diff
@@ -38,20 +39,27 @@ type Options struct {
 // Installer is entrypoint for installation process
 type Installer struct {
 	cluster *kubeoneapi.KubeOneCluster
+	secrets *kubeoneapi.KubeOneSecrets
 	logger  *logrus.Logger
 }
 
 // NewInstaller returns a new installer, responsible for dispatching
 // between the different supported Kubernetes versions and running the
-func NewInstaller(cluster *kubeoneapi.KubeOneCluster, logger *logrus.Logger) *Installer {
+func NewInstaller(cluster *kubeoneapi.KubeOneCluster, secrets *kubeoneapi.KubeOneSecrets, logger *logrus.Logger) *Installer {
 	return &Installer{
 		cluster: cluster,
+		secrets: secrets,
 		logger:  logger,
 	}
 }
 
 // Install run the installation process
 func (i *Installer) Install(options *Options) error {
+	_, err := credentials.ProviderCredentials(i.cluster.CloudProvider.Name, i.secrets)
```

Comment:
This is a hack to ensure validation would work. To be replaced with a proper runtime validation.